modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C

```diff
@@ -928,5 +948,135 @@ PolygonConcentricCircleMeshGeneratorBase::generate()
+
+        if (neighbor->neighbor_ptr(return_side) == elem)
+        {
+          neighbor->set_neighbor(return_side, nullptr);
+          boundary_info.add_side(neighbor, return_side, boundary_id);
+        }
+      }
+
+      mesh0->delete_elem(elem);
+    }
+
+    // build the 1d mesh from the new boundary
+    libMesh::ReplicatedMesh poly_mesh(comm());
+    auto side_list = mesh0->get_boundary_info().build_side_list();
+
+    Real min_side = std::numeric_limits<Real>::max();
+    std::unordered_map<dof_id_type, dof_id_type> old_new_node_map;
+    for (const auto & bside : side_list)
+    {
+      if (std::get<2>(bside) != boundary_id)
+        continue;
+
+      Elem * elem = mesh0->elem_ptr(std::get<0>(bside));
+      auto side = std::get<1>(bside);
+      std::unique_ptr<Elem> side_elem(elem->build_side_ptr(side));
+      auto copy = side_elem->build(side_elem->type());
+
+      Real l = side_elem->volume();
+      if (l < min_side)
+        min_side = l;
+
+      for (dof_id_type i : side_elem->node_index_range())
+      {
+        auto & n = side_elem->node_ref(i);
+
+        if (old_new_node_map.count(n.id()))
+          copy->set_node(i) = poly_mesh.node_ptr(old_new_node_map[n.id()]);
+        else
+        {
+          Node * node = poly_mesh.add_point(side_elem->point(i));
+          copy->set_node(i) = node;
+          old_new_node_map[n.id()] = node->id();
+        }
+      }
+      poly_mesh.add_elem(copy.release());
+    }
+    poly_mesh.skip_partitioning(true);
+    poly_mesh.prepare_for_use();
+
+    // triangulate with the 1d mesh
+    libMesh::Poly2TriTriangulator poly2tri(poly_mesh);
+    poly2tri.triangulation_type() = libMesh::TriangulatorInterface::PSLG;
+    poly2tri.set_interpolate_boundary_points(0);
+    poly2tri.set_refine_boundary_allowed(false);
+    poly2tri.set_verify_hole_boundaries(false);
+    const auto desired_area = getParam<Real>("inner_ring_desired_area");
+    const auto desired_area_func = getParam<std::string>("inner_ring_desired_area_func");
+    if (desired_area != 0)
+    {
+      poly2tri.desired_area() = desired_area;
+      if (desired_area_func != "")
+        paramError("inner_ring_desired_area_func",
+                   "cannot be specified with 'inner_ring_desired_area'");
+    }
+    else if (desired_area_func != "")
+    {
+      poly2tri.desired_area() = 0;
+      libMesh::ParsedFunction<Real> area_func{desired_area_func};
+      poly2tri.set_desired_area_function(&area_func);
+    }
+    else
+      poly2tri.desired_area() = min_side * min_side;
+    poly2tri.minimum_angle() = 0;
+    poly2tri.smooth_after_generating() = true;
+    poly2tri.elem_type() = libMesh::ElemType::TRI3;
+    // let us keep the center point
+    poly_mesh.add_point(libMesh::Point());
+    poly2tri.triangulate();
+    // keep the old subdomain id
+    for (auto elem : poly_mesh.element_ptr_range())
+      elem->subdomain_id() = _ring_block_ids[0];
```

Comment:
As mentioned before, `_ring_block_ids` is not necessarily non-empty as it is from an optional input parameter. `block_ids_new` is set.

Reply:
Changed to `block_ids_new`.